src/main/webapp/one-off/databaseInit.java

```diff
@@ -40,104 +39,123 @@
+import com.google.appengine.api.search.SearchServiceFactory;
+import com.google.appengine.api.search.StatusCode;
 import com.google.gson.Gson;
+import java.io.*;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import javax.servlet.ServletException;
-import java.util.*;
-import java.io.*;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Paths;
-import java.nio.file.Files;
+import org.apache.commons.lang3.StringUtils;
 
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/test")
 public class TestUploadServlet extends HttpServlet {
 
-    /**
-      * upload takes the several properties of a recipes, uses them in creating a new recipe entity and upload thi entity to the database
-     **/
-    public void upload(String  title, String imgURL, ArrayList<String> ingredients, ArrayList<String> stepList) throws Exception, IOException {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    
-        Entity recipeEntity = new Entity("Recipe");
+  /*
+   * Receives a recipe's properties and creates an entity, for the database and a document for the index
+   */
+  public void upload(
+    String title,
+    String imgURL,
+    ArrayList<String> ingredients,
+    ArrayList<String> stepList
+  )
+    throws Exception, InterruptedException, IOException {
+    IndexSpec indexSpec = IndexSpec
+      .newBuilder()
+      .setName("recipesIndex")
+      .build();
+    Index index = SearchServiceFactory.getSearchService().getIndex(indexSpec);
 
-        recipeEntity.setProperty("title", title);
-        recipeEntity.setProperty("index_title", title.toLowerCase());
-        recipeEntity.setProperty("imgURL", imgURL);
-        recipeEntity.setProperty("ingredients", ingredients);
-        recipeEntity.setProperty("stepList", stepList);
-        recipeEntity.setProperty("author", "Piece of Cake");
-        recipeEntity.setProperty("difficulty", "N/A");
-        recipeEntity.setProperty("prep_time", "N/A");
-        recipeEntity.setProperty("cook_time", "N/A");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    KeyRange keyRange = datastore.allocateIds("Recipe", 1L);
 
-        datastore.put(recipeEntity);
-    }
+    Entity recipeEntity = new Entity(keyRange.getStart());
+    recipeEntity.setProperty("title", title);
+    recipeEntity.setProperty("index_title", title.toLowerCase());
+    recipeEntity.setProperty("imgURL", imgURL);
+    recipeEntity.setProperty("ingredients", ingredients);
+    recipeEntity.setProperty("stepList", stepList);
+    recipeEntity.setProperty("author", "Piece of Cake");
```

Comment:
Please explain in a comment why we're defining an ID here. Also modularize this into a function called buildRecipeEntity.